README.md

```diff
@@ -6,46 +6,192 @@
+Let's see how this plugin can help you:
+
+### Where to find the plugin tab
+
+Git Machete IntelliJ Plugin is available under `Git` (`VCS` in older IntelliJ versions) tool window in the `Git Machete` tab.
+You can also use `Ctrl` + `Alt` + `Shift` + `M` shortcut to open it.
+
+![](docs/open_git_machete.gif)
 
-SLF4J logging in this plugin has the following categories:
+### Machete file edition
 
+`machete` file describes relation between branches in your repository (these relations are probably determined by order of branch creation - which branch form which - but this is not a rule).
+It's located inside `.git` directory in your repo.
+This is the example of `machete` file:
 ```
-binding
-branchlayout
-gitcore
-gitmachete.backend
-gitmachete.frontend.actions
-gitmachete.frontend.externalsystem
-gitmachete.frontend.graph
-gitmachete.frontend.ui
+develop
+    allow-ownership-link PR #123
+        build-chain PR #124
+    call-ws
+master
+    hotfix/add-trigger
 ```
+`develop`, `allow-ownership-link`, `build-chain`, `call-ws`, `master` and `hotfix/add-trigger` are branch names.
+Two of them (`allow-ownership-link` and `build-chain`) have custom annotation - it's some kind of short description for given branch and it can be any text that will be displayed next to this branch - in this case they are pull request numbers.
+Relation between these branches is determined by indentations - here single indent is 4 spaces but also tabulation(s) can be used.
+In the example above branches `allow-ownership-link` and `call-ws` are children of `develop` and `build-chain` is a child of `allow-ownership-link`.
+The same for `hotfix/add-trigger` and `master` - former is a child of latter.
+`develop` and `master` are root branches.
 
-The standard practice in Java logging is to use fully-qualified name of each class as category;
-in our case, however, we're only ever using the categories provided above.
-FQCN (and method name), however, is always a part of the log message itself.
+When branch tree is created, Git Machete indicates relation between parent branch and each of its child branches.
+If edge between them is green that means child branch is in sync with its parent branch - in other words there are no commits in parent branch that don't belong to the child.
+But if there are some of that commits, then edge is red - we need to [rebase](#rebase) child branch onto the parent.
 
-By default, IntelliJ logs everything with level `INFO` and above into `idea.log` file. <br/>
-The exact location depends on a specific IntelliJ installation; check `Help` -> `Show Log in Files` to find out. <br/>
-Tip: use `tail -f` to watch the log file as it grows.
+Machete file editor will help you to manage `machete` file: it underlines any errors (bad indentation or nonexistent branches) and propose branch names based on local repository branches.
+When file editing is done, you can click button in top right corner of file editor to refresh machete branch tree.
```

Comment:
Use `branch layout` consistently (ctrl+F across the project for `branch tree` maybe)